front/pages/api/w/[wId]/data_sources/[name]/index.ts

```diff
@@ -35,23 +35,17 @@ async function handler(
     });
   }
 
-  const dataSource = await DataSource.findOne({
-    where: auth.isUser()
-      ? {
-          workspaceId: owner.id,
-          visibility: {
-            [Op.or]: ["public", "private", "unlisted"],
-          },
-          name: req.query.name,
-        }
-      : {
-          workspaceId: owner.id,
-          // Do not include 'unlisted' here.
-          visibility: "public",
-          name: req.query.name,
-        },
-  });
+  if (!req.query.name || typeof req.query.name !== "string") {
+    return apiError(req, res, {
+      status_code: 404,
+      api_error: {
+        type: "data_source_not_found",
+        message: "The Data Source you requested was not found.",
+      },
+    });
+  }
 
+  const dataSource = await getDataSource(auth, req.query.name);
```

Comment:
we use this function first and then pull the model as it's much safer to have this logic in one place (even at the cost of 2 DB reads)